data-raw/eg.R

```diff
@@ -0,0 +1,247 @@
+# Dataset: eg
+# Description: Standard EG dataset from CDISC pilot study (includes random data)
+
+# Load necessary libraries
+library(dplyr)
+library(labelled)
+
+# Define constants and mappings
+studyid <- "CDISCPILOT01"
+domain <- "EG"
+usubjids <-
+  unique(vs$USUBJID) # Get unique subject IDs from vs data frame
+egtestcds <- c("QT", "HR", "RR", "ECGINT")
+visits <- c(
+  "SCREENING 1",
+  "SCREENING 2",
+  "BASELINE",
+  "AMBUL ECG PLACEMENT",
+  "WEEK 2",
+  "WEEK 4",
+  "AMBUL ECG REMOVAL",
+  "WEEK 6",
+  "WEEK 8",
+  "WEEK 12",
+  "WEEK 16",
+  "WEEK 20",
+  "WEEK 24",
+  "WEEK 26",
+  "RETRIEVAL"
+)
+egtpts <- c(
+  "AFTER LYING DOWN FOR 5 MINUTES",
+  "AFTER STANDING FOR 1 MINUTE",
+  "AFTER STANDING FOR 3 MINUTES"
+)
+visitnum_mapping <- c(
+  "SCREENING 1" = 1,
+  "SCREENING 2" = 2,
+  "BASELINE" = 3,
+  "AMBUL ECG PLACEMENT" = 3.5,
+  "WEEK 2" = 4,
+  "WEEK 4" = 5,
+  "AMBUL ECG REMOVAL" = 6,
+  "WEEK 6" = 7,
+  "WEEK 8" = 8,
+  "WEEK 12" = 9,
+  "WEEK 16" = 10,
+  "WEEK 20" = 11,
+  "WEEK 24" = 12,
+  "WEEK 26" = 13,
+  "RETRIEVAL" = 201
+)
+visitdy_mapping <- c(
+  "SCREENING 1" = -7,
+  "SCREENING 2" = -1,
+  "BASELINE" = 1,
+  "AMBUL ECG PLACEMENT" = 13,
+  "WEEK 2" = 14,
+  "WEEK 4" = 28,
+  "AMBUL ECG REMOVAL" = 30,
+  "WEEK 6" = 42,
+  "WEEK 8" = 56,
+  "WEEK 12" = 84,
+  "WEEK 16" = 112,
+  "WEEK 20" = 140,
+  "WEEK 24" = 168,
+  "WEEK 26" = 182,
+  "RETRIEVAL" = 168
+)
+egtptnum_mapping <- c(
+  "AFTER LYING DOWN FOR 5 MINUTES" = 815,
+  "AFTER STANDING FOR 1 MINUTE" = 816,
+  "AFTER STANDING FOR 3 MINUTES" = 817
+)
+egeltm_mapping <- c(
+  "AFTER LYING DOWN FOR 5 MINUTES" = "PT5M",
+  "AFTER STANDING FOR 1 MINUTE" = "PT1M",
+  "AFTER STANDING FOR 3 MINUTES" = "PT3M"
+)
+egtest_mapping <- c(
+  "QT" = "QT Duration",
+  "HR" = "Heart Rate",
+  "RR" = "RR Duration",
+  "ECGINT" = "ECG Interpretation"
+)
+egorresu_mapping <- c(
+  "QT" = "mmHg",
+  "HR" = "BEATS/MIN",
+  "RR" = "mmHg",
+  "ECGINT" = "LB"
+)
+egstresu_mapping <- c(
+  "QT" = "msec",
+  "HR" = "BEATS/MIN",
+  "RR" = "msec",
+  "ECGINT" = NA
+)
+
+# Extract unique date/time of measurements for each subject and visit
+egdtc <- vs %>%
+  select(USUBJID, VISIT, VSDTC) %>%
+  distinct() %>%
+  rename(EGDTC = VSDTC)
+
+# Set seed for reproducibility
+set.seed(123)
+
+# Function to generate patient data
+add_patient <- function(usubjid) {
```

Comment:
Do you even need a function here? What if you pass `USUBJID = unique(vs$USUBJID)` to do all patients at once?